cmake/sycl/Platform/Linux-AdaptiveCpp-SYCL.cmake

```diff
@@ -0,0 +1,31 @@
+# VecMem project, part of the ACTS project (R&D line)
+#
+# (c) 2022-2024 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# Use the standard GNU compiler options for hipSYCL.
+include( Platform/Linux-GNU )
+__linux_compiler_gnu( SYCL )
+include( Compiler/GNU )
+__compiler_gnu( SYCL )
+
+# Set up the dependency file generation for this platform. Note that SYCL
+# compilation only works with Makefile and Ninja generators, so no check is made
```

Comment:
Not sure I follow the logic that there's no check done for this reason. Is this checke elsewhere?

Reply:
Here: https://github.com/acts-project/vecmem/blob/main/cmake/sycl/CMakeDetermineSYCLCompiler.cmake#L7-L13

Basically only the (different flavours of)  Makefile and Ninja generators can be freely taught about new languages. Things like MSBuild, and Xcode can only deal with types that they internally already explicitly support. (Hence as I said in #300, technically it's possible to build `.metal` source files without teaching CMake how to build them. It's enough if Xcode knows how.)